lib/config/db/couchdb-get-vendor.js

```diff
@@ -1,30 +1,37 @@
-module.exports = checkVendor
+module.exports = getVendor
 
-var findKey = require('lodash').findKey
+var defaults = require('lodash').defaults
 
 var toCouchDbUrl = require('../../utils/pouchdb-options-to-couchdb-url')
 var removeAuth = require('../../utils/remove-auth-from-url')
 
-function checkVendor (state, couch, callback) {
+function getVendor (state, couch, callback) {
   couch({url: '/'}, function (error, response, data) {
     if (error || (response && response.statusCode !== 200)) {
       var url = toCouchDbUrl(state.db.options)
       return callback(new Error('Could not find CouchDB at ' + removeAuth(url)))
     }
 
-    var vendor = findKey(data, function (property) {
-      return /^welcome/i.test(property)
+    var vendor = defaults(data.vendor, {
+      type: getVendorType(data.vendor)
     })
 
-    if (vendor !== 'couchdb') {
+    if (vendor.type  !== 'couchdb') {
       state.server.log(
         ['database', 'warn'],
         'You are not running an official CouchDB distribution, ' +
-        'but "' + vendor + '". ' +
+        'but "' + vendor.name + '". ' +
         'This might not be fully supported. Proceed at your own risk.'
       )
     }
 
-    callback(null)
+    callback(vendor)
   })
 }
+
+
+function getVendorType (vendor) {
+  if(vendor.name === 'The Apache Software Foundation') return 'couchdb'
```

Comment:
I've just pushed up a change that might help here... I think there's two parts to vendor checking: ensuring the `welcome` value is set on a `couchdb` property, and then checking vendor (I've done this via name at present).

I've also just pushed tests for some of these scenarios... Coverage isn't 100% however.